dace/sdfg/analysis/schedule_tree/treenodes.py

```diff
@@ -113,46 +119,6 @@ def as_string(self, indent: int = 0):
         return indent * INDENTATION + f'assign {self.name} = {self.value.as_string}'
 
 
-@dataclass
-class ForScope(ControlFlowScope):
```

Comment:
I'm not sure we should remove those at all.
They serve different purposes than the control flow regions. What I think is that a ForScope should contain the controlflowregion

Reply:
They were not removed because of control flow regions, rather just because a single `GeneralLoopScope` (see line 157/123) took their functionality to have a closer 1:1 matching with the structure of the SDFG and the structured control flow tree. Or do they have an additional purpose over what the `GeneralLoopScope` provides?

Reply:
Initially I opted for simple classes, similar to AST definitions in other languages. Let’s leave it like this (since this is not an AST) and see how it works out. I would rename GeneralLoopScope to LoopScope, however.